pkg/providers/aws/cluster_network_provider.go

```diff
@@ -685,6 +687,58 @@ func (n *NetworkProvider) IsEnabled(ctx context.Context) (bool, error) {
 	return len(validBundledVPCSubnets) == 0, nil
 }
 
+// DeleteBundledCloudResources returns an error on any error deleting of the following resources
+// * elasticache subnet group
+// * rds subnet group
+// * ec2 security group
+//
+// it has been located under the cluster network provider as it requires 3 different aws sessions
+// (elasticache, rds and ec2) to delete the required resources even though it deals with bundled
+// resources. The majority of the functionality in this file relates to standalone aws vpc and it's
+// resources.
+func (n *NetworkProvider) DeleteBundledCloudResources(ctx context.Context) error {
+	logger := n.Logger.WithField("action", "deleteBundledCloudResources")
+
+	subnetGroupName, err := BuildInfraName(ctx, n.Client, "subnetgroup", DefaultAwsIdentifierLength)
+	if err != nil {
+		return errorUtil.Wrap(err, "error building bundle subnet group resource name on deletion")
+	}
+	logger.Infof("deleting bundled elasticcache subnet group %s, if it's not found it's already deleted and will continue", subnetGroupName)
+	_, err = n.ElasticacheApi.DeleteCacheSubnetGroup(&elasticache.DeleteCacheSubnetGroupInput{
+		CacheSubnetGroupName: aws.String(subnetGroupName),
+	})
+	if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() != elasticache.ErrCodeCacheSubnetGroupNotFoundFault {
+		return errorUtil.Wrap(err, "error deleting elasticache subnet group")
+	}
+	logger.Infof("deleting bundled rds subnet group %s, if it's not found it's already deleted and will continue", subnetGroupName)
+	_, err = n.RdsApi.DeleteDBSubnetGroup(&rds.DeleteDBSubnetGroupInput{
+		DBSubnetGroupName: aws.String(subnetGroupName),
+	})
+	if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() != rds.ErrCodeDBSubnetGroupNotFoundFault {
+		return errorUtil.Wrap(err, "error deleting rds subnet group")
+	}
```

Comment:
What if the error isn't an AWS err? e.g. `ok` is `false`? Maybe something like

```suggestion
	if awsErr, ok := err.(awserr.Error); !ok || awsErr.Code() != rds.ErrCodeDBSubnetGroupNotFoundFault {
		return errorUtil.Wrap(err, "error deleting rds subnet group")
	}
```